aequilibrae/paths/route_choice.py

```diff
@@ -0,0 +1,293 @@
+        self.psl_schema = RouteChoiceSet.psl_schema
+
+        self.compact_link_loads: Optional[np.array] = None
+        self.link_loads: Optional[np.array] = None
+        self.results: Optional[pa.Table] = None
+        self.where: Optional[pathlib.Path] = None
+
+        self.nodes = Optional[Union[List[int], List[Tuple[int, int]]]] = None
+
+    def set_algorithm(self, algorithm: str):
+        """
+        Chooses the assignment algorithm.
+        Options are, 'bfsle' for breadth first search with link removal, or 'link-penalisation'
+
+        'lp' is also accepted as an alternative to 'link-penalisation'
+
+        :Arguments:
+            **algorithm** (:obj:`str`): Algorithm to be used
+        """
+        algo_dict = {i: i for i in self.all_algorithms}
+        algo_dict["lp"] = "link-penalisation"
+        algo = algo_dict.get(algorithm.lower())
+
+        if algo is None:
+            raise AttributeError(f"Assignment algorithm not available. Choose from: {','.join(self.all_algorithms)}")
+
+        self.algorithm = algo
+        self._config["Algorithm"] = algo
+
+    def set_cores(self, cores: int) -> None:
+        """Allows one to set the number of cores to be used
+
+            Inherited from :obj:`AssignmentResultsBase`
+
+        :Arguments:
+            **cores** (:obj:`int`): Number of CPU cores to use
+        """
+        if not self.classes:
+            raise RuntimeError("You need load transit classes before overwriting the number of cores")
+
+        self.cores = cores
+
+    def set_paramaters(self, par: dict):
+        """
+        Sets the parameters for the route choice  TODO, do we want link specific values?
+
+        "beta", "theta", and "seed" are BFSLE specific parameters and will have no effect on link penalisation.
+        "penalty" is a link penalisation specific parameter and will have no effect on BFSLE.
+
+        Setting `max_depth`, while not required, is strongly recommended to prevent runaway algorithms.
+
+        - When using BFSLE `max_depth` corresponds to the maximum height of the graph of graphs. It's value is
+            largely dependent on the size of the paths within the network. For very small networks a value of 10
+            is a recommended starting point. For large networks a good starting value is 5. Increase the value
+            until the number of desired routes is being consistently returned.
+
+        - When using LP, `max_depth` corresponds to the maximum number of iterations performed. While not enforced,
+            it should be higher than `max_routes`. It's value is dependent on the magnitude of the cost field,
+            specifically it's related to the log base `penalty` of the ratio of costs between two alternative routes.
+
+
+        Parameter values can be scalars (same values for the entire network) or network field names
+        (link-specific values) - Examples: {'alpha': 0.15, 'beta': 4.0} or  {'alpha': 'alpha', 'beta': 'beta'}
```

Comment:
How (or which) parameters would be link-specific?

Reply:
Whoops, left one comments from a copy paste. We could have link specific parameters, say for link penalisation, but I'm not sure how useful that would be.